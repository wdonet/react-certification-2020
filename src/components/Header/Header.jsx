import React from 'react';
import { Navbar, Form, Col } from 'react-bootstrap';
import { HiMenu } from 'react-icons/hi';

import { Button } from '../Buttons/Button';
import InputText from '../TextInput/TextInput';
import ProfileImage from '../ProfileImage/ProfileImage';

function Header() {
  return (
    <Navbar bg="light" expand="lg">
      <Col sm={6}>
        <Button>
          <HiMenu />
        </Button>
        <InputText placeholder="Search" />
      </Col>
      <Col sm={{ offset: 3, span: 3 }}>
        <Form inline>
          <Form.Check type="switch" id="custom-switch" label="Dark mode" />
          <ProfileImage src="https://yt3.ggpht.com/ytc/AAUvwnighSReQlmHl_S_vSfvnWBAG5Cw4A0YxtE0tm5OpQ=s88-c-k-c0x00ffffff-no-rj" />
        </Form>
      </Col>
    </Navbar>
  );
}

export default Header;
