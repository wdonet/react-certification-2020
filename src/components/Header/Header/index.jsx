import React from 'react';
import Styled from './styled';
import './header.css';

import Logo from '../Logo';
import SearchBar from '../SearchBar';
import Login from '../Login';

const Header = () => {
  return (
    <Styled.Header>
      <Styled.LogoImage>
        <Logo />
      </Styled.LogoImage>
      <Styled.SearchArea>
        <SearchBar />
      </Styled.SearchArea>
      <Styled.Login>
        <Login />
      </Styled.Login>
    </Styled.Header>
  );
};

export default Header;
