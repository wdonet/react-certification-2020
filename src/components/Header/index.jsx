import React from 'react';
import Button from '../Button';
import Toolbar from '../Toolbar';
import IconButton from '../IconButton';
import { Menu } from '../Icons';

import { StyledHeader } from './styled';

const Header = () => {
  return (
    <StyledHeader>
      <Toolbar>
        <Button>
          <IconButton>
            <Menu width="24px" color="white" />
          </IconButton>
        </Button>
      </Toolbar>
    </StyledHeader>
  );
};

export default Header;
