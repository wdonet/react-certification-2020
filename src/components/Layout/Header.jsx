import React, { useContext, useRef } from 'react';
import styled, { ThemeContext } from 'styled-components';
import { TextField, IconWrapper, Avatar, Switch } from '../ui/index';
import hamburger from '../../assets/icons/hamburguer.png';
import defaultUser from '../../assets/icons/default_user.jpg';
import SearchContext from '../../providers/SearchContext';
import AppContext from '../../providers/AppContext';

const StyledHeader = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 100%;
  height: 64px;
  background-color: #00695c;
  overflow: hidden;
  position: fixed;
  top: 0;
`;

const StyledSection = styled.div`
  display: flex;
  align-items: center;
`;

const Header = () => {
  const { setHomeVideosView } = useContext(AppContext);
  const { switchTheme } = useContext(ThemeContext);
  const { search } = useContext(SearchContext);
  const ref = useRef(null);

  const setHomeVideosViewAndSearch = (query) => {
    setHomeVideosView();
    search(query);
  };

  return (
    <StyledHeader role="toolbar" data-testid="header">
      <StyledSection>
        <div>
          <IconWrapper role="button" src={hamburger} alt="hamburguer" />
        </div>
        <div data-testid="search-input">
          <TextField
            role="search"
            ref={ref}
            onKeyPress={({ charCode }) =>
              charCode === 13 && setHomeVideosViewAndSearch(ref.current.value)
            }
          />
        </div>
      </StyledSection>
      <StyledSection>
        <div data-testid="theme-mode-switch">
          <Switch onClick={switchTheme} />
        </div>
        <div data-testid="user-avatar">
          <Avatar role="button" alt="profile" src={defaultUser} />
        </div>
      </StyledSection>
    </StyledHeader>
  );
};

export default Header;
