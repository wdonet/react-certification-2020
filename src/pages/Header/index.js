export { default } from './Header.page';
