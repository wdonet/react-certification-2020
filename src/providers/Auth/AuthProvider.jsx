import React, { useState, useEffect, useCallback } from 'react';

import { AUTH_STORAGE_KEY } from '../../utils/constants';
import { loginUser } from '../../mocks/loginApi';
import { storage } from '../../utils/storage';

const AuthContext = React.createContext(null);

function AuthProvider({ authenticatedJest, children }) {
  const [authenticated, setAuthenticated] = useState(authenticatedJest);

  useEffect(() => {
    const lastAuthState = storage.get(AUTH_STORAGE_KEY);
    const isAuthenticated = Boolean(lastAuthState);

    setAuthenticated(isAuthenticated || authenticatedJest);
  }, []);

  const login = async ({ username, password }) => {
    try {
      const authorized = await loginUser({ username, password });
      if (authorized) {
        setAuthenticated(true);
        storage.set(AUTH_STORAGE_KEY, true);
      }
      return { success: true };
    } catch (error) {
      return { error };
    }
  };

  const logout = useCallback(() => {
    setAuthenticated(false);
    storage.set(AUTH_STORAGE_KEY, false);
  }, []);

  return (
    <AuthContext.Provider value={{ login, logout, authenticated }}>
      {children}
    </AuthContext.Provider>
  );
}

export { AuthContext };
export default AuthProvider;
