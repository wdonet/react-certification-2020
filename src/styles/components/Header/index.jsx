import React from 'react'
import PropTypes from 'prop-types'
import { StyledHeader, StyledMain, StyledSecondary, StyledCorner } from './styled'

const Header = ({ main, secondary, corner }) => {
  console.log(1)
  return (
    <StyledHeader data-testid="Header">
      <StyledMain>{main}</StyledMain>
      {secondary && <StyledSecondary>{secondary}</StyledSecondary>}
      {corner && <StyledCorner>{corner}</StyledCorner>}
    </StyledHeader>
  )
}

Header.propTypes = {
  main: PropTypes.node.isRequired,
  secondary: PropTypes.node,
  corner: PropTypes.node,
}

Header.defaultProps = {
  secondary: undefined,
  corner: undefined,
}

export default Header
