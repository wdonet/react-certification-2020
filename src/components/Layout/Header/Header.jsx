import React from 'react';
import { useHistory } from 'react-router-dom';
import { useAuth } from '../../../providers/Auth';
import { useYouTube } from '../../YouTube/YouTubeProvider';
import { Col, TitleLink, HeaderContainer, Search, Space, Switch } from './Header.styled';

const Header = () => {
  const { state, dispatch } = useYouTube();
  const { search = '', theme } = state;

  const { authenticated, logout } = useAuth();
  const history = useHistory();

  const setSearch = (event) => {
    dispatch({ type: 'search', payload: event.target.value });
  };

  return (
    <HeaderContainer>
      <TitleLink to="/">
        <Col>React Bootcamp</Col>
      </TitleLink>
      <Col>
        <Search placeholder="Search" value={search} onChange={setSearch} />
      </Col>
      <Space />
      <TitleLink to="/">
        <Col>Videos</Col>
      </TitleLink>
      <TitleLink to="/favorites">
        <Col>Favorites</Col>
      </TitleLink>
      <Col>
        <Switch
          onClick={() => {
            dispatch({ type: 'switchTheme' });
          }}
        >
          {theme === 'dark' ? 'Default' : 'Dark'}
        </Switch>
      </Col>

      <Col>
        <button
          type="button"
          onClick={() => {
            if (authenticated) {
              logout();
            } else history.push('/login');
          }}
        >
          {authenticated ? 'Logout' : 'Login'}
        </button>
      </Col>
    </HeaderContainer>
  );
};

export default Header;
