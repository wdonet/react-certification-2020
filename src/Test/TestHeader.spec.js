import React from 'react';
import { render, screen } from '@testing-library/react';
import AuthProvider from '../providers/Auth/Auth.provider';
import Header from '../components/Header';

describe('Header Components', () => {
  it('should take a snapshot', () => {
    const { asFragment } = render(
      <AuthProvider>
        <Header />
      </AuthProvider>
    );
    expect(
      asFragment(
        <AuthProvider>
          <Header />
        </AuthProvider>
      )
    ).toMatchSnapshot();
  });

  it('Expect the string Dark Mode', () => {
    render(
      <AuthProvider>
        <Header />
      </AuthProvider>
    );
    expect(screen.getByText('Dark Mode')).toBeTruthy();
  });

  it('Menu and logo are present', () => {
    render(
      <AuthProvider>
        <Header />
      </AuthProvider>
    );
    expect(screen.getAllByRole('img').length).toBe(2);
  });

  it('Input in header is present', () => {
    render(
      <AuthProvider>
        <Header />
      </AuthProvider>
    );
    expect(screen.getByRole('textbox')).toBeTruthy();
  });

  it('Toggle element is present in Header', () => {
    render(
      <AuthProvider>
        <Header />
      </AuthProvider>
    );
    expect(screen.getByRole('checkbox')).toBeTruthy();
  });
});
