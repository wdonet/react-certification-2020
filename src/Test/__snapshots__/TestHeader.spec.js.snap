// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Components should take a snapshot 1`] = `
<DocumentFragment>
  <header
    class="sc-bdfBwQ fnpANp"
  >
    <div
      class="sc-pFZIQ facKvC"
    />
    <img
      class="sc-jrAGrp yihMn"
      src="menu.png"
    />
    <div
      class="sc-dlfnbm cfbaly"
    >
      <input
        class="sc-gsTCUz cwBpJK"
        placeholder="Wizeline"
        type="text"
        value=""
      />
    </div>
    <div
      class="sc-fubCfw eiKTLY"
    >
      <img
        class="sc-gKsewC boIWSl"
        src="logo.png"
      />
      <div
        class="sc-hKgILt iCZJBc"
      >
        <div
          class="sc-iBPRYJ gRunDE"
        >
          Dark Mode
        </div>
        <input
          class="sc-jSgupP cGCuVX"
          id="checkbox"
          type="checkbox"
        />
        <label
          class="sc-eCssSg izygud"
          for="checkbox"
        />
      </div>
    </div>
  </header>
</DocumentFragment>
`;
